pyugrid/read_netcdf.py

```diff
@@ -150,9 +150,11 @@ def load_grid_from_nc_dataset(nc, grid, mesh_name=None, load_data=True):
                 raise ValueError(msg(defs['role']))
             continue
         except KeyError:
-            msg = ("File must include {} variables for {} "
-                   "named in mesh variable.").format
-            raise ValueError(msg(coord_names, defs['role']))
+            if defs['required']:
+                msg = ("File must include {} variables for {} "
+                       "named in mesh variable.").format
+                raise ValueError(msg(coord_names, defs['role']))
+            continue
```

Comment:
@ChrisBarker-NOAA can we be more "generic" here and catch both exceptions in one go?